src/Core/MakeModelMenu.elm

```diff
@@ -0,0 +1,348 @@
+
+                        _ ->
+                            ( RemoteData.Loading, Cmd.none )
+            in
+                ( { model | state = MakeSelection sortedFilteredResponse }, Cmd.none )
+
+        ModelsResponse response ->
+            let
+                ( sortedFilteredResponse, _ ) =
+                    case model.apiFilterField of
+                        "FactoryOrder" ->
+                            RemoteData.update (processRemoteData .factoryOrderAvailable) response
+
+                        "Stock" ->
+                            RemoteData.update (processRemoteData .stockAvailable) response
+
+                        "LeaseDeals" ->
+                            RemoteData.update (processRemoteData .leaseDealsAvailable) response
+
+                        _ ->
+                            ( RemoteData.Loading, Cmd.none )
+
+                newState =
+                    case model.state of
+                        ModelSelection make _ ->
+                            ModelSelection make sortedFilteredResponse
+
+                        _ ->
+                            model.state
+            in
+                ( { model | state = newState }, Cmd.none )
+
+        ModalBackClicked ->
+            let
+                newState =
+                    MakeSelection RemoteData.Loading
+            in
+                ( { model | state = newState }, getAvailableMakes (makesApiUrl model.apiEndpointUrl) )
+
+
+{-| Process the Make/Model data retrieved from research site
+-}
+processRemoteData :
+    (MenuItem a -> Bool)
+    -> List (MenuItem a)
+    -> ( List (MenuItem a), Cmd msg )
+processRemoteData field items =
+    let
+        sortField item =
+            item.name |> String.toUpper
+    in
+        ( items |> List.filter field |> List.sortBy sortField, Cmd.none )
+
+
+{-| A view representing the list of Makes/Models
+-}
+listView : (a -> List (Html msg)) -> WebData (List a) -> Html msg
+listView itemView itemsRemoteData =
+    case itemsRemoteData of
+        RemoteData.Success items ->
+            ul [ class "makes-models-menu__list list-unstyled" ]
+                (items
+                    |> List.map
+                        (\item ->
+                            li
+                                [ class "makes-models-menu__list-item" ]
+                                (itemView item)
+                        )
+                )
+
+        _ ->
+            spinnerView
+
+
+{-| A view representing a singular Make
+-}
+modalMakesView : WebData (List Make) -> Html Msg
+modalMakesView makesRemoteData =
+    let
+        makeView =
+            (\make ->
+                [ span [ class "makes-models-menu__link makes-models-menu__make", Html.Events.onClick (MakeSelected make) ]
+                    [ icon ("makes/" ++ String.join "_" (String.split "-" (String.toLower make.slug))) { size = "large", colour = "grey", colouring = "filled" }
```

Comment:
I suggest creating a makeIcon function in the `elm-theme`... very simple `makeIcon makeSlug`